pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -83,6 +83,19 @@ func TestGenerateVirtualHostDomains(t *testing.T) {
 			},
 			want: []string{"foo.local.campus.net", "foo.local.campus.net:80"},
 		},
+		{
+			name: "k8s service with default domain",
+			service: &model.Service{
+				Hostname:     "echo.default.svc.cluster.local",
+				MeshExternal: false,
+			},
+			port: 8123,
+			node: &model.Proxy{
+				DNSDomain: "default.svc.cluster.local",
+			},
+			want: []string{"echo.default", "echo.default.svc", "echo.default.svc.cluster.local",
```

Comment:
This is missing `echo` and `echo:8123`?

Reply:
sorry I realized my comment above may have been misleading. If the proxy is in `default` ns (as here), it should have `echo`, if its not it shouldn't

Reply:
Basically this PR should just remove things like `echo.default.svc.cluster` where we split the cluster domain

Reply:
I see. Let me fix that